pcawg_metadata_parser/parse_gnos_xml.py

```diff
@@ -1019,6 +1019,23 @@ def add_vcf_entry(donor, vcf_entry):
         else:
             donor.get('variant_calling_results').get('sanger_variant_calling')['is_output_and_tumour_specimen_counts_mismatch'] = False
 
+        # add the flags of is_bam_used_by_sanger_missing, is_normal_bam_used_by_sanger_missing, is_tumor_bam_used_by_sanger_missing
+        donor.get('variant_calling_results').get('sanger_variant_calling')['is_bam_used_by_sanger_missing'] = True
+        donor.get('variant_calling_results').get('sanger_variant_calling')['is_normal_bam_used_by_sanger_missing'] = True
+        donor.get('variant_calling_results').get('sanger_variant_calling')['is_tumor_bam_used_by_sanger_missing'] = True
+
+        # scan all the vcf input  under "variant_pipeline_input_info"
+        for vcf_input in donor.get('variant_calling_results').get('sanger_variant_calling').get('workflow_details').get('variant_pipeline_input_info'):
+           if vcf_input.get('attributes').get('analysis_id') == donor.get('normal_alignment_status').get('aligned_bam').get('gnos_id'): #check normal alignment
+               donor.get('variant_calling_results').get('sanger_variant_calling')['is_normal_bam_used_by_sanger_missing'] = False
+           else: # check the tumor
+               for tumor_alignment in donor.get('tumor_alignment_status'):
+                   if vcf_input.get('attributes').get('analysis_id') == tumor_alignment.get('aligned_bam').get('gnos_id'):
+                       donor.get('variant_calling_results').get('sanger_variant_calling')['is_tumor_bam_used_by_sanger_missing'] = False
```

Comment:
There is a problem here, it could be too soon to set the flag for missing tumor bam to False. Let's say, if there are two tumor specimen input BAMs, you find one is matching then set the flag to False. How about the other one? The second one could be missing!

Better way to do this is to set the flag to False initially, then set to True when failed to find a match. This way any of the input is missing the flag will be True.